keras_nlp/src/models/stable_diffusion_3/stable_diffusion_3_text_to_image.py

```diff
@@ -0,0 +1,146 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras import ops
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.stable_diffusion_3.stable_diffusion_3_backbone import (
+    StableDiffusion3Backbone,
+)
+from keras_nlp.src.models.stable_diffusion_3.stable_diffusion_3_text_to_image_preprocessor import (
+    StableDiffusion3TextToImagePreprocessor,
+)
+from keras_nlp.src.models.text_to_image import TextToImage
+
+
+@keras_nlp_export("keras_nlp.models.StableDiffusion3TextToImage")
+class StableDiffusion3TextToImage(TextToImage):
+    """An end-to-end Stable Diffusion 3 model for text-to-image generation.
+
+    This model has a `generate()` method, which generates image based on a
+    prompt.
+
+    Args:
+        backbone: A `keras_nlp.models.StableDiffusion3Backbone` instance.
+        preprocessor: A
+            `keras_nlp.models.StableDiffusion3TextToImagePreprocessor` instance.
+
+    Examples:
+
+    Use `generate()` to do image generation.
+    ```python
+    text_to_image = keras_nlp.models.StableDiffusion3TextToImage.from_preset(
+        "stable_diffusion_3_medium", height=512, width=512
+    )
+    text_to_image.generate(
+        "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k"
+    )
+
+    # Generate with batched prompts.
+    text_to_image.generate(
+        ["cute wallpaper art of a cat", "cute wallpaper art of a dog"]
+    )
+
+    # Generate with different `num_steps` and `classifier_free_guidance_scale`.
+    text_to_image.generate(
+        "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k",
+        num_steps=50,
+        classifier_free_guidance_scale=5.0,
+    )
+    ```
+    """
+
+    backbone_cls = StableDiffusion3Backbone
+    preprocessor_cls = StableDiffusion3TextToImagePreprocessor
+
+    def __init__(
+        self,
+        backbone,
+        preprocessor,
+        **kwargs,
+    ):
+        # === Layers ===
+        self.backbone = backbone
+        self.preprocessor = preprocessor
+
+        # === Functional Model ===
+        # TODO: Can we define the model here?
+        super().__init__(**kwargs)
+
+    def fit(self, *args, **kwargs):
+        raise NotImplementedError(
+            "Currently, `fit` is not supported for "
+            "`StableDiffusion3TextToImage`."
+        )
+
+    def generate_step(
+        self,
+        latents,
+        token_ids,
+        negative_token_ids,
+        num_steps,
+        classifier_free_guidance_scale,
+    ):
+        """A compilable generation function for batched of inputs.
+
+        This function represents the inner, XLA-compilable, generation function
+        for batched inputs.
+
+        Args:
+            latents: A <float>[batch_size, height, width, channels] tensor
```

Comment:
i don't think we use this notation usually? I'd just say "A float tensor with shape `(batch_size, height, width, channels)`..."

Reply:
There might be some legacy docstrings in the codebase:
https://github.com/keras-team/keras-hub/blob/9b1cf95d0a281ff54363af0d85953477fd55baf9/keras_hub/src/models/gemma/gemma_causal_lm.py#L328-L358

I adapted it from there. It's fixed now.